tiledb/sm/query/query_condition.h

```diff
@@ -128,8 +135,8 @@ class QueryCondition {
     /** The value data. */
```

Comment:
Honestly, I'm curious if there is a reason we have both a ByteVecValue and an UntypedDatum view? Don't they do the same thing? (store a contiguous array of bytes, with a given size and start pointer)

Reply:
+1 on this one. I think we should certainly simplify this class, it took me some time to figure out what all those `condition_value*` do. A few points:
1. We certainly need a member to persist the condition value, but unless I am missing something I don't see a reason to persist a view of it. 
2. A view (such as a `span<byte>`) can and should be constructed to be passed to functions if we just have a pointer to contiguous data and its size in hand, but if we already have a container (such as `ByteVecValue`) I guess we wouldn't need a view at all, right?
3. As for using `ByteVecValue`, I read in its docstring that : ` This class is only transitional. All uses should be rewritten to use ordinary types. Consider it deprecated at creation.` . So if we are refactoring, maybe it's worth considering replacing it with something more standard, i.e. a `vector` eg. `vector<byte>` ? @eric-hughes-tiledb  opinion on that?

Reply:
I wrote that comment. Before I turned it into a class `ByteVecValue`, that symbol was a typedef for `std::vector`. The comment I wrote at the time was right at the beginning of my engagement with type issues, before I had anything like a clear understanding what to do. It's not great, but it's better than it had been.

What's needed is a class, something like `UntypedValue` to hold an arbitrary single value for the relatively small but important cases where you don't have data in a table. The problem is that such a class, if written in isolation, wouldn't be all that useful, since it wouldn't provide much (if any) more functionality than `ByteVecValue` does. It's the combination of untyped data and type objects where the benefits would start to manifest. We don't have type objects yet, though. Since that's where type-specific dispatch and interpretation happens, value container classes don't do much.

At this time, for this PR, I'd recommend not tackling this issue as a local optimization.

Reply:
1 & 2. My original intention was just to use a reference to ByteVecValue for all comparisons.  But, the values that are being compared to can be "nullable" in which case we don't compare to the value stored in the ByteVecValue, but rather have to check a pointer being nullptr.  For that we use the pointer part of UntypedDataValue, which we can set to nullptr, since we can't get/set a nullptr with ByteVecValue.

Reply:
Since 1) `nullable` is a special case pretty much everywhere in the Query Condition code and logic, and 2) it was not clear to me reading `Clause` class that all the complexity comes from the need to support `nullables` : can we think of a clearer/more explicit way to handle this "special" case?
E.g. could we just introduce a `bool nullable` as an attribute in `Clause` so that we don't need the `view`, and all checks in the code become explicit to their intend?
Please let me know if I am still missing something.